pkg/util/util_linux.go

```diff
@@ -55,3 +55,42 @@ func DoesSocketExist(socketPath string) (bool, error) {
 	}
 	return false, nil
 }
+
+func CleanupFile(filePath string) error {
+	fileExists, err := DoesFileExist(filePath)
+	if err != nil {
+		return err
+	}
+	if fileExists {
+		if err := os.Remove(filePath); err != nil {
+			return fmt.Errorf("failed to remove stale file=%s with error: %+v", filePath, err)
+		}
+	}
+	return nil
+}
+
+func DoesFileExist(filePath string) (bool, error) {
+	info, err := os.Stat(filePath)
+	if err == nil {
+		return !info.IsDir(), nil
```

Comment:
this is for dealing what situation?

Reply:
it'd be enough to do `return true, nil` too, there's no specific situation why this would be checked with a directory

Reply:
Do we actually want to be more specific and check for a regular file type? https://pkg.go.dev/io/fs#FileMode

Reply:
Sounds good, I'm checking if the file is a regular file